sycl/test-e2e/Matrix/joint_matrix_out_bounds_impl.hpp

```diff
@@ -0,0 +1,117 @@
+           const auto sg_startx = global_idx - spmd_item.get_local_id(0);
+           const auto sg_starty = global_idy - spmd_item.get_local_id(1);
+
+           sub_group sg = spmd_item.get_sub_group();
+           joint_matrix<sub_group, bfloat16, use::a, TM, TK, layout::row_major>
+               sub_a;
+
+           // For B, since current implementation does not support non-packed
+           // layout, users need to specify the packed_b layout.
+           joint_matrix<sub_group, bfloat16, use::b, TK, TN,
+                        ext::intel::experimental::matrix::layout::packed>
+               sub_b;
+           joint_matrix<sub_group, float, use::accumulator, TM, TN> sub_c;
+           joint_matrix_fill(sg, sub_c, 1);
+           for (int k = 0; k < K; k += TK) {
+             joint_matrix_load(sg, sub_a, pA + (sg_startx * TM) * K + k, K);
+             // Assume we alreay in vnni format.
+             joint_matrix_load(sg, sub_b,
+                               pB + k * N + sg_starty / SG_SZ * TN * vnniFactor,
+                               N * vnniFactor);
+             sub_c = joint_matrix_mad(sg, sub_a, sub_b, sub_c);
+           }
+           joint_matrix_store(
+               sg, sub_c, pC + (sg_startx * TM) * N + sg_starty / SG_SZ * TN, N,
+               layout::row_major);
+         }); // parallel for
+   }).wait();
+}
+
+int main() {
+  static constexpr size_t MATRIX_M = 1024 + 14;
+  static constexpr size_t MATRIX_N = 1024;
+  static constexpr unsigned int vnniFactor = 2;
+
+  queue q;
+  bfloat16 *A = malloc_shared<bfloat16>(MATRIX_M * MATRIX_K, q);
+  bfloat16 *B = malloc_shared<bfloat16>(MATRIX_K * MATRIX_N, q);
+  bfloat16 *vnniB = malloc_shared<bfloat16>(MATRIX_K * MATRIX_N, q);
+  float *C = malloc_shared<float>(MATRIX_M * MATRIX_N, q);
+  float *D = malloc_shared<float>(MATRIX_M * MATRIX_N, q);
+
+  std::random_device dev;
+  std::uniform_real_distribution<float> fdistr(-1.0, 1.0);
+
+  for (int i = 0; i < MATRIX_M; i++) {
+    for (int j = 0; j < MATRIX_K; j++) {
+      A[i * MATRIX_K + j] = bfloat16(fdistr(dev));
+    }
+  }
+  for (int i = 0; i < MATRIX_K; i++) {
+    for (int j = 0; j < MATRIX_N; j++) {
+      B[i * MATRIX_N + j] = bfloat16(fdistr(dev));
+    }
+  }
+  for (int i = 0; i < MATRIX_M; i++) {
+    for (int j = 0; j < MATRIX_N; j++) {
+      C[i * MATRIX_N + j] = 1.0;
+      D[i * MATRIX_N + j] = 1.0;
+    }
+  }
+
+  matrix_vnni<bfloat16>(MATRIX_K, MATRIX_N, B, vnniB, vnniFactor);
+  matrix_multiply<float, bfloat16, MATRIX_M, MATRIX_K, MATRIX_K / vnniFactor,
+                  MATRIX_N * vnniFactor, MATRIX_M, MATRIX_N>(C, A, vnniB, q,
+                                                             vnniFactor);
+  matrix_multiply_ref(A, B, D, MATRIX_M, MATRIX_N, MATRIX_K);
+
+  bool res = true;
+  for (int i = 0; i < MATRIX_M; i++) {
+    for (int j = 0; j < MATRIX_N; j++) {
+      if ((fabs(C[i * MATRIX_N + j] - D[i * MATRIX_N + j])) > BF16_EPSILON) {
+        res = false;
+      }
+    }
+  }
```

Comment:
Can also be moved to common.hpp, something like matrix_compare
```suggestion
  bool res = matrix_compare(MATRIX_M, MATRIX_N, C, D);
```